### PR TITLE
Fix error when optional camel-cased props are missing

### LINKED ISSUE #2314: [Template validation props] Error when optional camel-cased props are missing
- [x] I have searched through existing issues
- [x] I have read through [docs](https://vuejs.github.io/vetur)
- [x] I have read [FAQ](https://github.com/vuejs/vetur/blob/master/docs/FAQ.md)

## Info

- Platform: macOS
- Vetur version: 0.28.0
- VS Code version: 1.49.1

## Problem

Vetur warn me that the props `true-value?` and `false-value?` are not provided, whereas those props are optional in the definition component.

image

**Toogle.vue**
image

### PR DESCRIPTION
**Pull request by yoyo930021**
Fixed #2314

### CODE REVIEW

File: test/interpolation/fixture/diagnostics/propTypeValidation/TSChild.vue [hunk 1]
```diff
@@ -8,7 +8,8 @@ export default Vue.extend({
     callback: {
       type: Function as PropType<() => void>
     },
-    arr: Array as PropType<string[]>
+    arr: Array as PropType<string[]>,
+    darkMode: { type: Boolean, default: false }
```
[the remark this answers is not in the document]
Comment (yoyo930021) [reply test/interpolation/fixture/diagnostics/propTypeValidation/TSChild.vue:12]: If add this camelCase props, it success fixed when no errors.

[the remark this answers is not in the document]
Comment (octref) [reply test/interpolation/fixture/diagnostics/propTypeValidation/TSChild.vue:12]: Just call it `camelCase` so it's clear what you are testing against

[the remark this answers is not in the document]
Comment (yoyo930021) [reply test/interpolation/fixture/diagnostics/propTypeValidation/TSChild.vue]: OK

**Comment by octref** [comment]
Awesome work 👏

**Pull request merged**